### PR TITLE
Switch from grpcio to tower-grpc.

### LINKED ISSUE #243: Migrate to tower-grpc for a Rust native gRPC implementation.
Post for context: https://medium.com/@siddontang/use-tower-grpc-for-tikv-6109cf8c61 and https://github.com/tikv/tikv/issues/3951

Let's investigate what it would take to migrate from `grpc-rs` to `tower-grpc`.

This would mean no more native toolset dependencies (CMake, C++ compiler, etc.) just to build Azure Functions for Rust.  Additionally, it may have performance benefits over `grpc-rs`.  We'll need some benchmarks around this.

I've already looked at the bidi support and it seems it meets the full-duplex needs of the Azure Functions protocol contract (unfortunately, no good samples, but the tests are well-written and understandable).

Related to #220.

### PR DESCRIPTION
**Pull request by peterhuene**
## What is the goal of this pull request?

To change the gRPC library to Tower gRPC.

## What does this pull request change?

This PR reimplements using Tower gRPC for the gRPC library.

As a result, this removes the dependency on CMake and a C++ compiler.

## What work remains to be done?

None.

## Do you consider it adequately tested?

Yes

## Related Issues

Closes #243.

## Notes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,8 +47,6 @@ pub fn greet(req: HttpRequest) -> HttpResponse {
 
 - [Installation](#installation)
   - [Requirements](#requirements)
-    - [CMake](#cmake)
-    - [C++ Compiler](#c-compiler)
```
Comment (t-eckert) [inline README.md]: 🎉🎊🎉

File: azure-functions/src/commands/run.rs [hunk 2]
```diff
@@ -4,27 +4,80 @@ use crate::{
...
+use tower_service::Service;
+use tower_util::MakeService;
 
 const UNKNOWN: &str = "<unknown>";
 
 thread_local!(static FUNCTION_NAME: RefCell<&'static str> = RefCell::new(UNKNOWN));
 
 type Sender = futures::sync::mpsc::UnboundedSender<StreamingMessage>;
 
+// TODO: replace with tower-request-modifier when published
```
Comment (t-eckert) [inline azure-functions/src/commands/run.rs]: Where can the progress on this be tracked?

Comment (t-eckert) [reply azure-functions/src/commands/run.rs]: Is it reasonable to include this change for 0.9.0?

File: azure-functions/src/commands/run.rs [hunk 3]
```diff
@@ -75,63 +128,57 @@ impl<'a> Run<'a> {
     pub fn execute(&self, mut registry: Registry<'static>) -> Result<(), String> {
         ctrlc::set_handler(|| {}).expect("failed setting SIGINT handler");
 
-        println!(
-            "Connecting to Azure Functions Host at {}:{}",
-            self.host, self.port
-        );
+        let uri: Uri = format!("http://{0}:{1}", self.host, self.port)
```
Comment (t-eckert) [inline azure-functions/src/commands/run.rs]: Is there a more descriptive name than `uri`? Or is the generality good?

Comment (t-eckert) [reply azure-functions/src/commands/run.rs]: Maybe `azure_functions_host`?

File: azure-functions/src/commands/run.rs (hunk 2 above)
Comment (peterhuene) [reply azure-functions/src/commands/run.rs]: I'll link the comment to https://github.com/tower-rs/tower-http/issues/24.  The Tower folks have been very accommodating already, so I'm not going to hold up 0.9.0 for tower-request-modifier to be published.  This implementation is a scaled down version of the request modifier that only does setting the origin.

File: azure-functions/src/commands/run.rs (hunk 3 above)
Comment (peterhuene) [reply azure-functions/src/commands/run.rs]: I think perhaps `host_uri` is best.  I'll make that change.

**Pull request merged**